pkg/cmd/miscellaneous.go

```diff
@@ -186,24 +186,35 @@ func getSeedNamespaceNameForShoot(shootName string) (namespaceSeed string) {
 	return shoot.Status.TechnicalID
 }
 
-// getProjectForShoot returns the Project for Shoot
-func getProjectForShoot() (v1beta1.Project, error) {
+// getProjectObject returns the Project for Shoot
+func getProjectObject() (v1beta1.Project, error) {
 	var target Target
 	project := v1beta1.Project{}
 	ReadTarget(pathTarget, &target)
 	Client, err := clientToTarget("garden")
 	checkError(err)
-	if target.Target[1].Kind == "project" {
-		projectName := target.Target[1].Name
+
+	// user for case garden-project-shoot
+	if isTargeted("project", "shoot") {
+		projectName, err := getTargetName("project")
+		checkError(err)
 		p, err := Client.GardenerV1beta1().Projects().Get(projectName, metav1.GetOptions{})
 		checkError(err)
 		project = *p
-	} else {
+	}
+
+	// user for case garden-seed-shoot
+	if isTargeted("seed", "shoot") {
```

Comment:
change this to else-if, otherwise it could happen that you unnecessarily try to fetch the project even if you have already fetched it. I'm not sure if both project and seed can be targeted at the same time, but anyhow, an else-if will make prevent unnecessary calls

Reply:
done, add Placeholder and pring warmng and os.exit(2) so far.

Reply:
I do not understand what you mean by that, can you elaborate

Reply:
some experiment forget to remove it

Reply:
To progress quicker on this PR, please take a look at my suggestion for the getProjectObject function. I feel there are a few things that are wrong or missing and need to be improved which I have addressed with my suggestion:
- added else-if (as I already suggested above)
- getRole check is missing in case the seed is targeted 
- checking for shoot on `if isTargeted("project", "shoot") {` is unecessary, I have removed it: `if isTargeted("project") {`
- no error was returned in case the project or seed and shoot was not targeted
- misleading error message in case len(shootList.Items) == 0

and I have adapted the signature to return a project pointer, which means you also have to adapt the callers of course
```go
// getProjectObject returns the Project for Shoot
func getProjectObject() (*v1beta1.Project, error) {
	var target Target
	ReadTarget(pathTarget, &target)
	Client, err := clientToTarget("garden")
	checkError(err)

	if isTargeted("project") {
		projectName, err := getTargetName("project")
		checkError(err)
		return Client.GardenerV1beta1().Projects().Get(projectName, metav1.GetOptions{})
	} else if isTargeted("seed", "shoot") {
		if getRole() == "user" {
			return nil, errors.New("can't determine project. Target project instead of seed")
		}
		seedName, err := getTargetName("seed")
		checkError(err)
		shootName, err := getTargetName("shoot")
		checkError(err)

		shootList, err := Client.GardenerV1beta1().Shoots(metav1.NamespaceAll).List(metav1.ListOptions{
			FieldSelector: fields.SelectorFromSet(
				fields.Set{
					core.ShootSeedName: seedName,
					"metadata.name":    shootName,
				}).String(),
		})
		checkError(err)
		if len(shootList.Items) == 0 {
			return nil, errors.New("No shoot found with name " + shootName + " that is running on the seed " + seedName)
		} else if len(shootList.Items) > 1 {
			return nil, errors.New("There are multiple shoots with the name " + shootName + " that are running on the seed " + seedName)
		}
		projectNamespace := shootList.Items[0].Namespace

		projectList, err := Client.GardenerV1beta1().Projects().List(metav1.ListOptions{})
		checkError(err)

		for _, p := range projectList.Items {
			if *p.Spec.Namespace == projectNamespace {
				return &p, nil
			}
		}
	}

	return nil, errors.New("can't determine project")
}
```
I have just wrote down this code to reflect the open points. I have not tested it..

Reply:
`return nil, errors.New("No shoot found with name " + shootName + " that is running on the seed " + seedName)` 

`cannot convert nil (untyped nil value) to v1beta1.Project` I am use os.Exit(2) now. 
let me know if there other options will change it

Reply:
I cannot reproduce it.

have you noted this statement?:
> and I have adapted the signature to return a project pointer, which means you also have to adapt the callers of course

-> have you changed the function signature to be the same as in my suggestion?

Reply:
Thanks, you are right, I am not familiar function signature will take a look